src/cast.c

```diff
@@ -2597,7 +2641,180 @@ bool isVoidArrayLiteral(Expression *e, Type *other)
         ((ArrayLiteralExp *)e)->elements->dim == 0);
 }
 
+/**
+ * This function should be called by iterateAliasThis.
+ * It is needed to collect of subtypes of the expression type.
+ * It checks if type hasn't been found and returns it if yes.
+ */
+static bool atCollectSubtypes(Scope *sc, Expression *e, void *ctx, Expression **outexpr)
+{
+    Expressions *candidates = (Expressions *)ctx;
+    bool duplicate = false;
+    assert(e->type);
+    for (int j = 0; j < candidates->dim; ++j)
+    {
+        if ((*candidates)[j]->type->equals(e->type))
+        {
+            duplicate = true;
+            break;
+        }
+    }
+    if (!duplicate)
+    {
+        *outexpr = e;
+    }
+    return false;
```

Comment:
Yes, it should collect all subtypes, and it shouldn't break the loop by returning true.